ledger/participant-integration-api/src/main/scala/platform/store/FlywayMigrationsBase.scala

```diff
@@ -0,0 +1,294 @@
+  //      database
+  //   4. Delete all migrations used in step 1
+  //   5. Update the value below
+  protected def baselineVersion: String
+
+  protected def getDataSource: Future[DataSource]
+
+  protected def locations: List[String]
+
+  private def runF[T](t: FluentConfiguration => Future[T]): Future[T] =
+    getDataSource.flatMap(dataSource => t(configurationBase(dataSource)))
+
+  private def run[T](t: FluentConfiguration => T): Future[T] =
+    runF(fc => Future(t(fc)))
+
+  private def configurationBase(dataSource: DataSource): FluentConfiguration =
+    Flyway
+      .configure()
+      .locations(locations: _*)
+      .dataSource(dataSource)
+
+  private def getStatus(flyway: Flyway): FlywayStatus = {
+    val info = flyway.info()
+    val current = info.current()
+    val pending = info.pending()
+    val applied = info.applied()
+
+    def isBeforeBaseline(m: MigrationInfo) = !m.getVersion.isAtLeast(baselineVersion)
+
+    // The Flyway history is immutable.
+    // To delete a migration, a new row is inserted into the history table that marks the migration as deleted.
+    // We are looking for rows that have not been marked as deleted by other rows (not DELETED),
+    // and that are not themselves deletion markers (not DELETE).
+    val notDeletedMigrationsBeforeBaseline = applied.toList
+      .filter(m =>
+        isBeforeBaseline(
+          m
+        ) && m.getType != MigrationType.DELETE && m.getState != MigrationState.DELETED
+      )
+      .map(_.getVersion.getVersion)
+
+    if (current == null) {
+      FlywayStatus.DatabaseEmpty(info.pending().length)
+    } else if (isBeforeBaseline(current)) {
+      FlywayStatus.SchemaBeforeBaseline(current.getVersion.getVersion, baselineVersion)
+    } else if (notDeletedMigrationsBeforeBaseline.nonEmpty) {
+      FlywayStatus.NeedsBaselineReset(notDeletedMigrationsBeforeBaseline)
+    } else if (pending.nonEmpty) {
+      FlywayStatus.Outdated(current.getVersion.getVersion, applied.length, pending.length)
+    } else {
+      FlywayStatus.NoMigrationsPending(current.getVersion.getVersion, applied.length)
+    }
+  }
+
+  private def resetBaseline(flyway: Flyway): Unit = {
+    val info = flyway.info()
+
+    assert(
+      info.current().getVersion.isNewerThan(baselineVersion),
+      "Current version is below the baseline, can't reset the baseline",
+    )
+    assert(
+      info.applied().exists(m => !m.getVersion.isNewerThan(baselineVersion)),
+      "No migrations below baseline found, baseline reset not required",
+    )
+
+    def migrationsToString(
+        migrations: java.util.List[org.flywaydb.core.api.output.RepairOutput]
+    ): String = {
+      if (migrations.isEmpty) {
+        "<none>"
+      } else {
+        migrations.asScala.map(_.version).mkString(", ")
+      }
+    }
+
+    // Expected output is:
+    // - All migrations below the baseline are DELETED
```

Comment:
Could you align the order in comment with the order in the code?
Would it make sense to check these expected invariants in the code and at least print an error when e.g. number of removed migrations > 0?